api/segments/models.py

```diff
@@ -198,6 +198,38 @@ def deep_clone(self) -> "Segment":
 
         return cloned_segment
 
+    def match_rules_to_segment(self, segment: "Segment") -> None:
+        _rules = self.rules.all()
+        matched_rules = set()
+        matched_sub_rules = set()
+
+        for rule in segment.rules.all():
+            sub_rules = rule.rules.all()
+            for sub_rule in sub_rules:
+                sub_rule_matched = False
+                for _rule in _rules:
+                    if sub_rule_matched:
+                        break
+
+                    if _rule in matched_rules and _rule.version_of != rule:
+                        continue
+
+                    if rule.type != _rule.type:
+                        continue
+                    for _sub_rule in _rule.rules.all():
+                        if _sub_rule in matched_sub_rules:  # pragma: no cover
```

Comment:
Why do we need the pragma no cover here?

Reply:
I've altered a test to expand coverage to tackle it.